install/src/command.rs

```diff
@@ -1171,13 +1172,24 @@ pub fn init_or_update(config_file: &str, is_init: bool, check_only: bool) -> Res
         release_dir.join("solana-release"),
         config.active_release_dir(),
     )
-    .map_err(|err| {
-        format!(
+    .map_err(|err| match err.raw_os_error() {
+        #[cfg(windows)]
+        Some(os_err) => if os_err == winapi::shared::winerror::ERROR_PRIVILEGE_NOT_HELD {
+            "You need to run this command with administrator privileges.".to_string()
+        } else {
+            format!(
+                "Unable to symlink {:?} to {:?}: {}",
+                release_dir,
+                config.active_release_dir(),
+                err
+            )
+        },
```

Comment:
You can do match with a conditional avoid the copying of the fallback error, ie:

```suggestion
        #[cfg(windows)]
        Some(os_err) if os_err == winapi::shared::winerror::ERROR_PRIVILEGE_NOT_HELD => {
            "You need to run this command with administrator privileges.".to_string()
        },
```

More info at https://doc.rust-lang.org/reference/expressions/match-expr.html#match-guards

Reply:
Thank you very much, this is so important. 

In fact, I started developing with windows, but I found that these codes could not be compiled correctly on windows because some build.rs would report errors.

Reply:
![image](https://github.com/anza-xyz/agave/assets/6522980/9d6666a0-eac9-429a-890c-8604a85a9ea5)